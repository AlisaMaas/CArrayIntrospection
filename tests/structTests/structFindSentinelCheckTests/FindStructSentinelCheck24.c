#include "structHeader.h"
int foo (struct foo *f, int b) {
    char *p = "HELLO";
    if (!f->string)
       f->string = "THIS IS MORE PLACEHOLDER";
    if (b) {
      p = f->string;
    }
    int i;
    for (int i = 0; p[i]; i++) {
        if(p[i] == 'a')
            return i;
    }
    return i;
}