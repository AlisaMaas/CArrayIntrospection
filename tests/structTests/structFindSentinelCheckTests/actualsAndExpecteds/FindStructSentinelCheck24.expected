Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array':
Analyzing function: foo
	We found: 1 loops
	Examining struct struct.foo[0] in loop for.cond
		There are 1 sentinel checks of this element in this loop
			We cannot bypass all sentinel checks for this element in this loop.
		Sentinel checks: 
			for.cond
